src/loguru_parallel/propagate.py

```diff
@@ -24,15 +24,9 @@ def wrapped_func(*args, **kwargs):
         child_logger._core = parent_logger._core
         return func(*args, **kwargs)
 
+    wrapped_func = wrap_non_picklable_objects(wrapped_func, keep_wrapper=False)
     try:
-        wrapped_func = functools.wraps(func)(
-            wrap_non_picklable_objects(wrapped_func, keep_wrapper=False)
-        )
+        wrapped_func = functools.wraps(func)(wrapped_func)
     except AttributeError:
         " functools.wraps fails on some callable objects "
```

Comment:
After all, why was it safe to continue processing even if an `AttributeError` occurs?